src/contexts/StatsContext.tsx

```
@@ -0,0 +1,442 @@
+        effectiveType: connection?.effectiveType,
+        downlink: connection?.downlink,
+        optimizedTimeout: shouldOptimizeForMobile ? 15000 : 30000,
+      });
+    };
+
+    detectNetworkConditions();
+
+    // Re-detect on network change if supported
+    if ('connection' in navigator) {
+      const connection = (navigator as unknown as { connection?: NetworkInformation }).connection;
+      if (connection && connection.addEventListener) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)